amundsen_rds/models/table.py

```diff
@@ -0,0 +1,173 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+from sqlalchemy import (
+    BigInteger, Boolean, Column, ForeignKey, Integer, String, Text
+)
+from sqlalchemy.orm import relationship
+
+from amundsen_rds.models.base import (
+    KEY_LEN, NAME_LEN, PUBLISHED_TAG_LEN, URL_LEN, Base
+)
+
+
+class Table(Base):
+    """
+    Table model.
+    """
+    __tablename__ = 'table_metadata'
+
+    rk = Column(String(KEY_LEN), primary_key=True)
+    name = Column(String(NAME_LEN), nullable=False)
+    is_view = Column(Boolean, nullable=False)
+    schema_rk = Column(String(KEY_LEN), ForeignKey('schema_metadata.rk', ondelete='cascade'), nullable=False)
+    published_tag = Column(String(PUBLISHED_TAG_LEN), nullable=False)
+    publisher_last_updated_epoch_ms = Column(BigInteger)
+
+    application = relationship('Application', uselist=False, secondary='application_table')
+    schema = relationship('Schema', uselist=False)
+    columns = relationship('TableColumn', order_by='TableColumn.sort_order')
+    description = relationship('TableDescription', order_by='TableDescription.rk', uselist=False)
+    programmatic_descriptions = relationship('TableProgrammaticDescription', order_by='TableProgrammaticDescription.rk')
+    tags = relationship('Tag', order_by='Tag.rk', secondary='table_tag')
+    badges = relationship('Badge', order_by='Badge.rk', secondary='table_badge')
+    timestamp = relationship('TableTimestamp', uselist=False)
+    watermarks = relationship('TableWatermark', order_by='TableWatermark.rk')
+    source = relationship('TableSource', uselist=False)
+    usage = relationship('TableUsage', order_by='TableUsage.user_rk', backref='table')
+    readers = relationship('User', order_by='User.rk', secondary='table_usage', backref='tables_read')
+    owners = relationship('User', order_by='User.rk', secondary='table_owner', backref='tables_owned')
+    followers = relationship('User', order_by='User.rk', secondary='table_follower', backref='tables_followed')
+
+
+class TableDescription(Base):
+    """
+    Description model for table.
+    """
+    __tablename__ = 'table_description'
+
+    rk = Column(String(KEY_LEN), primary_key=True)
+    description_source = Column(String(32), nullable=False)
+    description = Column(Text)
+    table_rk = Column(String(KEY_LEN), ForeignKey('table_metadata.rk', ondelete='cascade'), nullable=False)
+    published_tag = Column(String(PUBLISHED_TAG_LEN))
+    publisher_last_updated_epoch_ms = Column(BigInteger)
+
+
+class TableProgrammaticDescription(Base):
+    """
+    Programmatic description model for table.
+    """
+    __tablename__ = 'table_programmatic_description'
+
+    rk = Column(String(KEY_LEN), primary_key=True)
+    description_source = Column(String(32), nullable=False)
+    description = Column(Text)
```

Comment:
qq: does text have no length limit? also does it have any difference between using text vs blob for type?

Reply:
There is no length specified here because some backend does not accept a length argument for 'Text'. For Blob, we can use sqlachelmy LargeBinary to set Blob for a column in MySQL. Regarding the difference, Text and LargeBinary size limit would probably be dependent on the backend database. For MySQL, both should be up to 64kb. The other difference between Text and Blob should be the way of comparison and sorting, whether based on the character set or numeric value of bytes, according to the mysql doc.